### DISCUSSION TITLE
EDI x12 850 to 856

### DISCUSSION THREAD

**Discussion by EarlWer** [blueseerERP/blueseer#139] [Q&A] [answered]
Do you have any automated processes to convert an 850 into an 856?
I am looking to convert my old EDI system to something more modern... 
Java is not my first choice but an Open Source tool that I can use and expand is at the top of that list.
I looks like BOTS is no longer being developped, so here I am ;-)

**Comment by vaughnte** [comment, marked as answer]
A sample map was just created for this scenario.   It's available in the downloads for version 7.0.12 and higher.   A video tutorial is also available for it's usage here:  https://youtu.be/cXaZtsjmaPk .

**Reply by EarlWer** [reply]
Nice video explanation, thanks. I am looking for something a bit more realistic:

Incoming 850 mapped to Order. I think you already have that.  Complications: SDQ segments are Store/Qty breakdowns for each PO1 line. So if they order 1000 widgets, there may be SDQ segments with a breakdown of how many for each store. That means you need a Store/Qty table under the OrderDetails. 

Order to Shipment: This is the useful / complicated part. The Inventory file must have info like Qty per box. With that, I should be able to click on a 'Ship' or 'Invoice' (you already have) button that will create the Shipment.  You may already do that BUT the Shipment tables must include multiple pallets and/or cartons!  If they order 1000 widgets and the Qty. Per Box = 20, the Shipment should have 50 boxes with 20 widgets in each one OR use the SDQ Store/Qty breakdown to create a set of boxes for each store.
... but wait!  There's more:  If SDQ and/or multiple items, the process should try to fill each box as much as possible.
This way, we tell the shipper what to pack into each carton. The alternative is scan and pack which is very slow.

Note: Many EDI companies will pay from the ASN/856, so there is no INV/810 to send...

Shipment to ASN/856: This is actually simple IF the Shipment tables have the correct structure. Shipment, Pallet (optional), Carton, Contents. 

My system does this and a lot more...  but it's based on a very old database engine (DataEase) and limited to a 32-bit client.

So, the front-end is really the key... which you have written with Swing.... and I have zero experience with it.
Sorry to be so verbose...  I'm trying to find a more modern environment for my system.

**Reply by vaughnte** [reply]
The sample 850 to 856 translation map is just that...a transformation of 850 data to 856 data....at a generic level...to be used by whomever to construct more complex mapping scenarios.   Your description fits in the category of complexity...but also specific to your business process...and therefore outside the scope of a generic offering which is what this BlueSeer map provides.   However, you can take BlueSeer as-is and customize the map (with custom DB lookups) to construct any SOTI or SOPI ASN structure one can dream of...but it would be tailored to a specific business process...such as your own.    Your SDQ store/qty breakdown is a good example...but I would add a 3rd key....store/item/qty so that any packaging hiearchary would be store/item/qty specific.   BlueSeer's current ship_mstr and ship_det tables are built on item level detail....but a SDQ store/item/qty relationship could be queried to build the appropriate SOTI or SOPI structure.   Note that the BlueSeer Mapping tool is a generic mapping engine...independent of business use-case rules.   Given your description of your process...it might be easier for you to just build an app tailored toward your process....the language or toolset is being irrelevant at this point...as long as the app achieves the desired goal.    Incidentally, you could probably use the BlueSeer code_mstr table (a two key / value) table to store your SDQ store/qty lookup info in....then build a 'very' custom map from the sample template provided.   There is also a DB -to- 856 generic map as well...which is probably the one I would use given the amount of customization you'll need to do to get your ASN structure that you want.
